### ISSUE TITLE
placement_configs order mismatch triggers node pool recreation

### ISSUE THREAD

**Issue by kviktorman** [terraform-providers/terraform-provider-oci#1045]
### Terraform Version and Provider Version

```
Terraform v0.12.24
+ provider.oci v3.72.0
```

### Affected Resource(s)

* data "oci_core_subnets"
* resource "oci_containerengine_node_pool"

### Terraform Configuration Files

```hcl
data "oci_core_subnets" "subnet_list" {
  compartment_id = var.target_deploy_compartment
  vcn_id         = oci_core_virtual_network.oci_test_virtual_network.id
}

resource "oci_containerengine_node_pool" "test_node_pool" {
  cluster_id         = oci_containerengine_cluster.test_cluster.id
  compartment_id     = var.target_deploy_compartment
  kubernetes_version = var.cluster.kubernetesVersion
  name               = var.cluster.kubernetesNodePoolName
  node_shape         = var.cluster.kubernetesNodeShape
  node_config_details {
    dynamic "placement_configs" {
      for_each = [
        for subnet in data.oci_core_subnets.subnet_list.subnets : {
          availabilityDomain = subnet.availability_domain
          subnetId           = subnet.id
      }]

      content {
        availability_domain = placement_configs.value.availabilityDomain
        subnet_id           = placement_configs.value.subnetId
      }
    }
    size = var.cluster.kubernetesNodeNumber
  }

  node_source_details {
    image_id    = var.cluster.kubernetesNodeImageOCID
    source_type = "IMAGE"
  }
}

output "debug_output" {
  value = data.oci_core_subnets.subnet_list
}
```

### Debug Output

```
subnet_ids          = [
    "ocid1...a", // This is for AD-3
    "ocid1...b", // This is for AD-1
    "ocid1...c", // This is for AD-2
]

~ node_config_details {
    size = 1

  ~ placement_configs {
      ~ availability_domain = "...-AD-1" -> "...-AD-3"
      ~ subnet_id           = "ocid1...b" -> "ocid1...a"
    }
    placement_configs {
        availability_domain = "...-AD-2"
        subnet_id           = "ocid1...c"
    }
  ~ placement_configs {
      ~ availability_domain = "...-AD-3" -> "...-AD-1"
      ~ subnet_id           = "ocid1...b" -> "ocid1...c"
    }
}
```
### Expected Behavior

I think one of the 2 should be the behavior:
* the placement_configs order should not change after the resource has been created
* the placement_configs should be ordered before comparison so they match with the order for an existing resource

### Actual Behavior

We have availability zone dedicated network. When we create the node pool for the Kubernetes cluster we query directly from the "oci_core_subnets" the available subnets. We loop through the subnets and create the placement_configs dynamically.

What we experience that every time we run 'terraform apply' the node pool resource is marked, that it needs recreation. This is due the order mismatch of the "placement_configs".

Thanks for your help with my issue.

### Steps to Reproduce

Run terraform apply 2 times after each other.

1. `terraform apply`
2. `terraform apply`

### Important Factoids

So far what I noticed is that the "oci_core_subnets" returns the subnets in the same order. It seems that the "placement_configs" of a created resource are ordered differently, most probably alphabetically based on the `availability_domain` and not alphabetically based on the `subnet_id`.

### References

Regards to the use case you can find more in the [24557 issue](https://github.com/hashicorp/terraform/issues/24557) tracked in the core Terraform repository.

**Comment by jotruon** [comment]
Hi @kviktorman, 
Thanks for reporting this bug.
The root cause for this miss order of `placement_configs` is the service return a sorted List of `placement_configs` so the next apply will show a diff.

The fix can be changing the typeList to typeSet for  `placement_configs`:
```
"placement_configs": {
							Type:     schema.TypeList, -> schema.TypeSet
```
We are working on the fix now

**Comment by kviktorman** [comment]
I just noticed that in the bug report, I miss typed the last value.
For  "...-AD-3" -> "...-AD-1" the subnet is "ocid1...a" -> "ocid1...b", because "ocid1...c" is AD2
Sorry for that and thanks a lot!

**Comment by jotruon** [comment]
@kviktorman a workaround for this is provide a sorted `placement_configs` for `oci_containerengine_node_pool`.
We are working on the fix.

**Comment by jotruon** [comment]
Hi @kviktorman , 
The fix for this is on v3.75.0
Thanks
